quality_of_service_demo/rclcpp/src/message_lost_talker.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rcutils/cmdline_parser.h"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+
+#include "sensor_msgs/msg/image.hpp"
+
+#include "quality_of_service_demo/visibility_control.h"
+
+using namespace std::chrono_literals;
+
+namespace quality_of_service_demo
+{
+void print_usage()
+{
+  std::cout <<
+    "Usage:\n"
+    "message_lost [-h]\n\n"
+    "optional arguments:\n"
+    "-h:                           Print this help message.\n"
+    "-s <message_size>:            Message size in KiB, default to 8192 KiB" <<
+    std::endl;
+}
+
+class MessageLostTalker : public rclcpp::Node
+{
+public:
+  QUALITY_OF_SERVICE_DEMO_PUBLIC
+  explicit MessageLostTalker(const rclcpp::NodeOptions & options)
+  : Node("message_lost_talker", options),
+    message_size_(8u * 1024u * 1024u)  // 8MB
+  {
+    const std::vector<std::string> & args = this->get_node_options().arguments();
+    if (args.size()) {
+      // Argument usage
+      if (args.cend() != std::find(args.cbegin(), args.cend(), "-h")) {
+        print_usage();
+        throw std::invalid_argument("asked for help");
```

Comment:
Instead of throwing, consider exiting the program, perhaps with a TODO like this other demo: 

https://github.com/ros2/demos/blob/bda392ed2e4ce4c9ad428935382e97502ba7c676/image_tools/src/cam2image.cpp#L49-L51

Reply:
The problem with `exit` is that it doesn't unwind the stack, and thus the program will leak.
I agree that throwing an exception isn't nice either.
I will think if there's another option, if not I will use exit.

>  // TODO(jacobperron): Replace with a mechanism for a node to "unload" itself 
>  // from a container. 

Though having a way to unload a component sounds nice, this is an easier case: failing to load a node.
Maybe we need an specific exception for this, the component manager could catch it and terminate more gracefully.

Reply:
> Maybe we need an specific exception for this, the component manager could catch it and terminate more gracefully.

Sure. It might also make sense for the component manager to catch exceptions from components in general and fail gracefully (which it doesn't seem to be doing at the moment).

Reply:
I've used std::exit, and added a todo

Reply:
You've fixed the cases for `-s`, but I think we should also use `std::exit(0)` here too (for `-h`).

Reply:
Thanks! I've fixed that